src/librustc_codegen_utils/link.rs

```diff
@@ -61,7 +62,7 @@ pub fn find_crate_name(sess: Option<&Session>,
                        attrs: &[ast::Attribute],
                        input: &Input) -> String {
     let validate = |s: String, span: Option<Span>| {
-        cstore::validate_crate_name(sess, &s, span);
+        creader::validate_crate_name(sess, &s, span);
```

Comment:
For example, could this move to `rustc` or to a different crate? The `rustc_metadata` crate is quite heavyweight and pulling it in for one function may not quite be worth it